core/common/src/main/java/alluxio/util/CommonUtils.java

```diff
@@ -426,18 +431,34 @@ public void remove() {
 
   /**
    * Executes the given callables, waiting for them to complete (or time out).
-
+   *
    * @param callables the callables to execute
    * @param <T> the return type of the callables
+   * @throws TimeoutException if one of the callables times out
+   * @throws Exception if one of the callables throws an exception
    */
-  public static <T> void invokeAll(List<Callable<T>> callables) {
+  public static <T> void invokeAll(List<Callable<T>> callables) throws TimeoutException, Exception {
```

Comment:
In the case one of the callables throws an exception, there is no guarantee that all the callables have run before returning?